app/javascript/lib/scp-api.js

```diff
@@ -31,11 +33,36 @@ if (
   })
 }
 
+/**
+ * Populates global id-to-name map of retrieved filters, for easier analytics.
+ * See downstream use of window.SCP.filterNamesById in metrics-api.js.
+ */
```

Comment:
To the extent we need such a map, it should be maintained in the SearchFacetContext in SearchFacetProvider.js or just the individual FacetControl that grabbed them.  We shouldn't be adding globals unless we need to access them in non-React parts of the app.

Reply:
This global variable was a bad code smell indeed.  I've refactored that to not be global, and moved it and its surrounding function as described in my [other comment](https://github.com/broadinstitute/single_cell_portal_core/pull/454#discussion_r393842055).